### PR TITLE
fix bug with interleaving dataset reader

### PR DESCRIPTION
**Pull request by epwalsh**
Fixes https://github.com/allenai/allennlp/discussions/4933#discussioncomment-606859.

### CODE REVIEW

File: allennlp/data/dataset_readers/interleaving_dataset_reader.py [hunk 1]
```diff
@@ -52,6 +59,24 @@ def __init__(
...
+        """
+        super()._set_worker_info(info)
+        for reader in self._readers.values():
+            reader._set_worker_info(info)
+
+    @overrides
+    def _set_distributed_info(self, info: Optional[DistributedInfo]) -> None:
+        """
+        Should only be used internally.
+        """
```
Comment (dirkgr) [inline allennlp/data/dataset_readers/interleaving_dataset_reader.py:75]: I think if the method starts with an `_`, and there is nothing good to say in the docstring, just don't have the docstring. `_` methods don't go into the docs anyways, right?

**Review by dirkgr — APPROVED** [review]

File: allennlp/data/dataset_readers/interleaving_dataset_reader.py (hunk 1 above)
Comment (epwalsh) [reply allennlp/data/dataset_readers/interleaving_dataset_reader.py:75]: Yea, I think I accidentally copy and pasted this from the base class. Will remove.

File: allennlp/data/dataset_readers/interleaving_dataset_reader.py [hunk 2]
```diff
@@ -52,6 +59,24 @@ def __init__(
             raise ConfigurationError(f"invalid scheme: {scheme}")
         self._scheme = scheme
 
+    @overrides
+    def _set_worker_info(self, info: Optional[WorkerInfo]) -> None:
+        """
+        Should only be used internally.
+        """
```
Comment (epwalsh) [inline allennlp/data/dataset_readers/interleaving_dataset_reader.py:66]:
```suggestion
```

File: allennlp/data/dataset_readers/interleaving_dataset_reader.py (hunk 1 above)
Comment (epwalsh) [inline allennlp/data/dataset_readers/interleaving_dataset_reader.py:75]:
```suggestion
```

**Pull request merged**